engine/src/flutter/shell/platform/darwin/ios/framework/Source/UITextInput+TextRange.swift

```diff
@@ -0,0 +1,226 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import UIKit
+
+@objc(FlutterTextPosition)
+final class TextPosition: UITextPosition, NSCopying {
+  let offset: UInt
+  @objc let affinity: UITextStorageDirection
+
+  init(index: UInt, affinity: UITextStorageDirection = .forward) {
+    self.offset = index
+    self.affinity = affinity
+  }
+
+  override func isEqual(_ object: Any?) -> Bool {
+    guard let other = object as? TextPosition else {
+      return false
+    }
+    // Affinity is only visual.
+    return offset == other.offset
+  }
+
+  func copy(with zone: NSZone? = nil) -> Any {
+    return TextPosition(index: offset, affinity: affinity)
+  }
+
+  override var description: String {
+    "TextPosition(\(index))"
```

Comment:
is this only for debugging? maybe we can remove it

Reply:
Hmm what would be the concerns for overriding `description`? Would that increase the binary size? If it's not used in production code the compiler should dead function eliminate it (along with the string data) I think?

Reply:
Oh it's overriding an objc method so it could be called via dynamic dispatch, so the compiler probably can't eliminate this. 

I think I figured out why it was added: in tests the messages with TextRange / TextPosition in them won't be very human readable without the `description` overrides, when the tests fail.

Reply:
(so it's needed in test and I couldn't find any documentation that says`debugDescription` will be treated differently from `description` by the compiler so I'll keep it as-is).